tests/Unit/NumericalAlgorithms/Interpolation/Test_MultiLinearSpanInterpolation.cpp

```diff
@@ -0,0 +1,146 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t dim, size_t num_var>
+void test() {
+  //  CAPTURE(Degree);
+
+  // pseudo random number generator
+  std::random_device rd;   // seed
+  std::mt19937 gen(rd());  // generator
+
+  std::uniform_real_distribution<> dist_func(-1., 1.);
+  std::uniform_int_distribution<size_t> dist_int(10, 20);
+
+  // Main idea: The table performs a multi-linear interpolation
+  // Each element in the table will be multi-linear function,
+  // so we can use this directly as a test.
+  // For simplicity, we will use the table to globally reproduce a
+  // single multi-linear function.
+
+
+  std::array< std::array<double, dim>, num_var> A;
```

Comment:
we prefer longer descriptive names for variables. E.g. `table` or even `matrix_a` and `vector_b`. This at the very least just makes find-replace and searching easier in the file :)